src/StackExchange.Redis/Interfaces/IDatabase.cs

```diff
@@ -325,6 +325,153 @@ public interface IDatabase : IRedis, IDatabaseAsync
         /// <remarks><seealso href="https://redis.io/commands/hexists"/></remarks>
         bool HashExists(RedisKey key, RedisValue hashField, CommandFlags flags = CommandFlags.None);
 
+
+        /// <summary>
+        /// Set the remaining time to live in milliseconds for a field of hash
+        /// After the timeout has expired, the field of the hash will automatically be deleted.
+        /// </summary>
+        /// <param name="key">The key of the hash.</param>
+        /// <param name="hashField">The field in the hash to set expire time.</param>
+        /// <param name="expiry">The timeout to set.</param>
+        /// <param name="when">under which condition the expiration will be set using <see cref="ExpireWhen"/>.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns> null if the key does not exist. Otherwise returns the result of operation for given field;
+        ///		2: field deleted because the specified expiration time is due
+        ///		1: expiration time set/updated
+        /// 	0: expiration time is not set/update (a specified ExpireWhen condition is not met)
+        ///		-2 : no such field
+        /// </returns>
+        ExpireResult? HashFieldExpire(RedisKey key, RedisValue hashField, TimeSpan expiry, ExpireWhen when = ExpireWhen.Always, CommandFlags flags = CommandFlags.None);
```

Comment:
In what scenario would this return null?

Reply:
when the key does not exist, it returns in compliance with response from server